src/plot_api/plot_api.js

```diff
@@ -245,6 +246,7 @@ Plotly.plot = function(gd, data, layout, config) {
 
     function drawAxes() {
         // draw ticks, titles, and calculate axis scaling (._b, ._m)
+        if(gd._fullLayout.xaxis && gd._fullLayout.xaxis.rangeslider) RangeSlider.draw(gd);
```

Comment:
**Important**. `RangeSlider.draw` should get call regardless of what's in `fullLayout`, to allow `RangeSlider.draw` to also remove existing sliders from the DOM. e.g.

```js
Plotly.relayout(Tabs.getGraph(), 'xaxis.rangeslider.visible', false);
```

does not work at the moment.

The logic above, where you join `[0]` to the range slider layer, should be generalized so that when `rangeslider.visible` is set to `false`, `[]` should be join to the DOM layer.